java/client/src/main/java/com/youtube/vitess/client/Proto.java

```diff
@@ -203,6 +210,26 @@ public EntityId apply(Map.Entry<byte[], ?> entry) {
         // Float, Double
         this.type = Query.Type.FLOAT64;
         this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Date) {
+        // Date
+        this.type = Query.Type.DATE;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Time) {
+        // Time
+        this.type = Query.Type.TIME;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Timestamp) {
+        // DateTime, TimeStamp
+        this.type = Query.Type.TIMESTAMP;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Boolean ) {
+        // Boolean
+        this.type = Query.Type.BIT;
+        this.value = ByteString.copyFromUtf8(((boolean)value) ? "1" : "0");
+      } else if (value instanceof BigDecimal) {
+        // BigDecimal
+        this.type = Query.Type.FLOAT64;
```

Comment:
The problem with that is the precision can change.
Ex. BigDecimal(0.23), BigDecimal.toPlanString() will result in 0.2300000000000000099920072216264088638126850128173828125
In MySQL the maximum scale is 30.
So if a column is defined as my_col decimal(2,30) then this will result in wrong data in the database.